docs/source/testing/integration.rst

```diff
@@ -18,4 +18,164 @@
 Writing Integration Tests for NWChemEx
 ######################################
 
-TODO: Write me!!!!
+NWChemEx is a modular ecosystem designed with separation of concerns as a key
+design point. An example of this separation can be found with the 
+self-consistent field (SCF) code and the integrals code. These two components of
+NWX are linked by PluginPlay and SimDE and are intended to be used together, but
+neither is therefore explicitly required for the development of the other and 
+therefore reside in separate Plugins. The unit tests for these libraries are 
+intended to ensure basic functionality and correctness, which can usually be 
+accomplished with simple test data that allow the unit tests to run quickly. 
+With that said, the initial development and testing of the SCF becomes very 
```

Comment:
It could be useful to draw some dependency diagrams. For example show that because plugins are parallel to one another they can't depend on each other, i.e., 

```mermaid
flowchart TD
    SimDE --> SCF
    SimDE --> Integrals
    SimDE --> ChemCache
    SCF --> NWChemEx
    Integrals --> NWChemEx
    ChemCache --> NWChemEx
```

However since tests are on top of the plugin itself we can also have the tests depend on NWChemEx, i.e.,
```mermaid
flowchart TD
    SCFTest[SCF Tests]
    IntegralsTest[Integrals Tests]
    ChemCacheTest[ChemCache Tests]
    SimDE --> SCF
    SimDE --> Integrals
    SimDE --> ChemCache
    SCF --> NWChemEx
    Integrals --> NWChemEx
    ChemCache --> NWChemEx
    SCF --> SCFTest
    Integrals --> IntegralsTest
    ChecmCache --> ChemCacheTest
    NWChemEx --> SCFTest
    NWChemEx --> IntegralsTest
    NWChemEx --> ChemCacheTest
```